Objects/rangeobject.c

```diff
@@ -915,12 +908,12 @@ get_len_of_range(long lo, long hi, long step)
 static PyObject *
 fast_range_iter(long start, long stop, long step, long len)
 {
-    rangeiterobject *it = PyObject_New(rangeiterobject, &PyRangeIter_Type);
+    _PyRangeIterObject *it = PyObject_New(_PyRangeIterObject, &PyRangeIter_Type);
     if (it == NULL)
         return NULL;
-    it->start = start;
-    it->step = step;
-    it->len = len;
+    it->start = Py_SAFE_DOWNCAST(start, long, sdigit);
```

Comment:
Would it be cleaner to change the signature of `fast_range_iter` to take sdigits, rather than longs?

Reply:
I'm not so sure -- downcasts need to happen somewhere, and `fast_range_iter` is called in two places. The `PyLong_AsLong` calls could theoretically be replaced with `Py_SIZE` checks and `ob_item[0]` accesses, but I think the existing code is fine using the public API.